client/fresheyes-bot/src/util.ts

```diff
@@ -39,67 +41,135 @@ function formatTime(arg: string) {
 
-  const authors = new Set(
-    arg
-      .map((item) => item?.user?.login as string)
-      .filter((author) => {
-        const isPrAuthor = author.toLowerCase() === prAuthor.toLowerCase();
-        const isDrahtBot = author.toLowerCase() === "drahtbot".toLowerCase();
+  const userTypeCompare = (item: Record<string, any>, type: string) =>
+    item?.user?.type.toLowerCase() === type.toLowerCase();
 
-        if (isBitcoinBot) {
-          return !isPrAuthor && !isDrahtBot;
-        }
+  const isBitcoinBot = arg.some(
+    (item) => userLoginCompare(item, "DrahtBot") || userTypeCompare(item, "bot")
+  );
 
-        return !isPrAuthor;
-      })
-  ).size;
+  const isAuthorPresent = arg.some((item) => userLoginCompare(item, prAuthor));
+
+  const isRegularAuthor = (item: Record<string, any>) =>
+    !userLoginCompare(item, prAuthor) &&
+    !userLoginCompare(item, "DrahtBot") &&
+    !userTypeCompare(item, "bot");
 
-  const comments = arg.length;
-  const commentText = comments === 1 ? "comment" : "comments";
-  const reviewersText = authors === 1 ? "reviewer" : "reviewers";
+  const authors = new Set(
+    arg.filter(isRegularAuthor).map((item) => item.user.login.toLowerCase())
+  ).size;
+  const uniqueBots = new Set(
+    arg
+      .filter(
+        (item) =>
+          userTypeCompare(item, "bot") ||
+          (userLoginCompare(item, "DrahtBot") && item.body.trim() !== "")
+      )
+      .map((item) => item.user.login.toLowerCase())
+  ).size;
 
-  const botComment = isBitcoinBot ? "and 1 bot" : "";
+  const nonBotCommentCount = arg.filter(
+    (item) => isRegularAuthor(item) && item.body.trim() !== ""
+  ).length;
+
+  const allNonEmptyCommentsCount = arg.filter(
+    (item) => item.body.trim() !== ""
+  ).length;
+
+  const isReviewWithoutComment = nonBotCommentCount === 0 && authors >= 1;
+
+  const commentText = allNonEmptyCommentsCount === 1 ? "comment" : "comments";
+  const reviewersText =
+    isReviewWithoutComment || authors === 0
+      ? ""
+      : authors === 1
+      ? "reviewer"
+      : "reviewers";
+
+  const botCommentText = uniqueBots === 1 ? "1 bot" : `${uniqueBots} bots`;
+  const botComment = isBitcoinBot
+    ? isAuthorPresent
+      ? `, ${botCommentText}`
+      : isReviewWithoutComment || authors === 0
+      ? botCommentText
+      : `and ${botCommentText}`
+    : "";
+
+  const authorText =
+    isReviewWithoutComment || authors === 0 ? "" : ` ${authors}`;
+
+  const issueBody = `There ${
+    allNonEmptyCommentsCount <= 1 ? "was" : "were"
+  } ${allNonEmptyCommentsCount} ${commentText} left by${authorText} ${reviewersText}${botComment} ${
+    isAuthorPresent ? "and the author" : ""
+  } for this pull request`;
 
   return [
     {
-      body: `There were ${comments} ${commentText} left by ${authors} ${reviewersText} ${botComment} for the pull request`,
-      created_at: arg[0].created_at,
+      body: issueBody,
+      created_at: arg[0]?.created_at || new Date().toISOString(),
```

Comment:
This solves a critical point of failure in the code when there are no issue comments for the recreated pr. What if we put an explicit return at the top of the code block where the function definition starts for better error handling ?

Reply:
Fine by me. Should I make a commit for that or do you want to do it?

Reply:
I'll add it from here